ffmuc-mesh-vpn-wireguard-vxlan/luasrc/lib/gluon/upgrade/400-mesh-vpn-wireguard

```diff
@@ -15,14 +23,14 @@ uci:delete_all('wireguard', 'wireguard', function(peer)
 	return peer.preserve ~= '1'
 end)
 
+-- TODO: fix get_bool(network.wg_mesh.disabled), currently always true if field is empty
 local mesh_enabled = uci:get_bool('gluon', 'mesh_vpn', 'enabled') -- default
 	or uci:get_bool('fastd', 'mesh_vpn', 'enabled') --migration
-	or not uci:get_bool('network', 'wg_mesh', 'disabled') --compatiblity with upstream
+	-- or not uci:get_bool('network', 'wg_mesh', 'disabled') --compatiblity with upstream
 	or wg_enabled -- specific config
 
 uci:section("wireguard", "wireguard", "mesh_vpn", {
 	iface = site.mesh_vpn.wireguard.iface(),
-	limit = site.mesh_vpn.wireguard.limit(),
 	broker = site.mesh_vpn.wireguard.broker(),
 	enabled = mesh_enabled,
```

Comment:
It would be much cooler to just react to `gluon.mesh_vpn.enabled='1'` like upstream does too? :)

Reply:
Aren't we?

Reply:
you are once reading the value from gluon.mesh_vpn.enabled on upgrade but use
`mesh_vpn_enabled="$(uci get wireguard.mesh_vpn.enabled)"`  in L53 to check if the vpn is enabled.
So users can't just use `uci set gluon.mesh_vpn.enabled=0` to turn off wireguard.

But instead have to tinkle with
`uci get wireguard.mesh_vpn.enabled`
or am I missing something?

Reply:
No, you are correct. I was not aware that `gluon.mesh_vpn` is still being used.

Reply:
@grische tried to "fix" this in #99 though we found out, that the current way this is set is correct as the two settings are synced in:

https://github.com/freifunk-gluon/community-packages/blob/33bfda10cc357e59e2b0df9b3a190fd7123c7c0c/ffmuc-mesh-vpn-wireguard-vxlan/luasrc/usr/lib/lua/gluon/mesh-vpn/provider/wireguard.lua#L19-L22

Just like it is done for the upstream package:

https://github.com/freifunk-gluon/gluon/blob/d4847b697c1fb9da44751d659072e77e4dae6967/package/gluon-mesh-vpn-wireguard/luasrc/usr/lib/lua/gluon/mesh-vpn/provider/wireguard.lua#L19